src/raster_types.jl

```diff
@@ -12,7 +13,24 @@ function Raster(dataset::GDALDatasetH)
                     _getrastercount(dataset)) # number of bands
     finalizer(raster, closeraster)
     raster
-end 
+end
+
+function Base.show(io::IO,raster::Raster)
+     if (raster.dataset == C_NULL)
+         println(io, raster.dataset)
+         return null;
```

Comment:
Nope. This part is only to make the code snippet self-contained so you could run it. Please note that since `_bandtype` does not go in the PR, its reference within `show` needs to be changed back to `dtype(_getrasterband`.

You can push another commit to the same branch on GitHub and the PR will be updated automatically. Since there are 4 (soon to be 5) partial commits that add up to one atomic change, you would need to `squash` (in git lingo, tutorials abound if you search for it) the commits into a single commit before it would be merged, so you could use this as a chance to try that if you haven't before, or you could submit a new PR this time if you prefer.